src/devices/bus/spectrum/zxbus.h

```diff
@@ -0,0 +1,124 @@
+    *13      D7 │ 3│   │ 3│ +5V  (guaranteed +5V only here ! )
+     BLK (CSDS) │ 4│   │ 4│ DOS (DCDOS)
+     -not-used- │ 5│   │ 5│ KAY = 14Mhz  / SCORPION = +12V
+    *14      D0 │ 6│   │ 6│ GND
+    *15      D1 │ 7│   │ 7│ GND
+    *12      D2 │ 8│   │ 8│ CLK Z80 (SCORPION aka /RAS 3.5Mhz ONLY ! )
+    *10      D6 │ 9│   │ 9│ A0      *30
+    *9       D5 │10│   │10│ A1      *31
+    *8       D3 │11│   │11│ A2
+    *7       D4 │12│   │12│ A3
+              - │13│   │13│ IOGE  (IORQCE)
+    *17     NMI │14│   │14│ GND
+              - │15│   │15│ RDR(RDROM)   (CSROMCE)
+    *19    MREQ │16│   │16│ RS (BIT_4 OF #7FFD)
+    *20    IORQ │17│   │17│ -not-used-
+    *21      RD │18│   │18│ -not-used-
+    *22      WR │19│   │19│ BUSRQ
+     -not-used- │20│   │20│ RES     *26
+    *24    WAIT │21│   │21│ A7      *37
+     -not-used- │22│   │22│ A6      *36
+     -not-used- │23│   │23│ A5      *35
+    *27      M1 │24│   │24│ A4
+          RFRSH │25│   │25│ CSR  (CSROM)
+    *38      A8 │26│   │26│ BUSAK
+    *40     A10 │27│   │27│ A9      *39
+  KAY=+5V  n.u. │28│   │28│ A11     *1
+  KAY=+12V n.u. │29│   │29│ n.u. KAY=+5V
+            GND │30│   │30│ GND
+                │  │   │  │
+                │ B│   │A │
+                └──┴───┴──┘
+  '-' - not used
+  '*' - Z80 CPU out
+
+*******************************************************************************************/
+
+#ifndef MAME_BUS_SPECTRUM_ZXBUS_H
+#define MAME_BUS_SPECTRUM_ZXBUS_H
+
+#pragma once
+
+class zxbus_slot_device : public device_t, public device_slot_interface
+{
+public:
+	zxbus_slot_device(machine_config const &mconfig, char const *tag, device_t *owner, uint32_t clock);
+
+	template <typename T, typename U>
+	zxbus_slot_device(machine_config const &mconfig, char const *tag, device_t *owner, u32 clock, T &&zxbus_tag, U &&slot_options, const char *default_option, bool fixed)
+		: zxbus_slot_device(mconfig, tag, owner, clock)
+	{
+		option_reset();
+		slot_options(*this);
+		set_default_option(default_option);
+		set_fixed(fixed);
+        m_zxbus_bus.set_tag(std::forward<T>(zxbus_tag));
+	}
+
+protected:
+    zxbus_slot_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	virtual void device_start() override;
+
+	required_device<device_t> m_zxbus_bus;
+};
+
+DECLARE_DEVICE_TYPE(ZXBUS_SLOT, zxbus_slot_device)
+
+
+class zxbus_device : public device_t
+{
+public:
+    zxbus_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
+	template <typename T> void set_iospace(T &&tag, int spacenum) { m_iospace.set_tag(std::forward<T>(tag), spacenum); }
+	template<typename T> void install_device(offs_t addrstart, offs_t addrend, T &device, void (T::*map)(class address_map &map), uint64_t unitmask = ~u64(0))
+	{
+		m_iospace->install_device(addrstart, addrend, device, map, unitmask);
+	}
+
+	void add_slot(const char *tag);
+	void add_slot(device_slot_interface *slot);
```

Comment:
Is there a reason for the variant that takes a tag?  Can you just have a single `void add_slot(zxbus_slot_device &slot);` to simplify things?